src/tests/Interop/MarshalAPI/FunctionPointer/FunctionPointer.cs

```diff
@@ -111,4 +113,145 @@ public static void RunGetDelForOutIntTest()
         }
         Assert.Equal(expectedValue, outVar);
     }
+
+    [UnmanagedCallersOnly]
+    static int UnmanagedExportedFunction(float arg)
+    {
+        return Convert.ToInt32(arg);
+    }
+    
+    [UnmanagedCallersOnly]
+    static BlittableGeneric<int> UnmanagedExportedFunctionBlittableGenericInt(float arg)
+    {
+        return new() { X = Convert.ToInt32(arg) };
+    }
+
+    [UnmanagedCallersOnly]
+    static BlittableGeneric<string> UnmanagedExportedFunctionBlittableGenericString(float arg)
+    {
+        return new() { X = Convert.ToInt32(arg) };
+    }
+
+    class GenericCaller<T>
+    {
+        internal static unsafe T GenericCalli<U>(void* fnptr, U arg)
+        {
+            return ((delegate* unmanaged<U, T>)fnptr)(arg);
+        }
+
+        internal static unsafe BlittableGeneric<T> WrappedGenericCalli<U>(void* fnptr, U arg)
+        {
+            return ((delegate* unmanaged<U, BlittableGeneric<T>>)fnptr)(arg);
+        }
+    }
+
+    struct BlittableGeneric<T>
+    {
+        public int X;
+    }
+
+    [Fact]
+    public static void RunGenericFunctionPointerTest()
+    {
+        Console.WriteLine($"Running {nameof(RunGenericFunctionPointerTest)}...");
+        int outVar = 0;
+        int expectedValue = 42;
```

Comment:
We are not able to test `int.MaxValue` as the purpose here is to test whether the return value matches `Convert.ToIne32(float)`, but `Convert.ToIne32((float)int.MaxValue)` will fail because `(float)int.MaxValue = 2.1474836E+09` which exceeds `int.MaxValue`

Reply:
@hez2010 That makes sense. My intent here was that we should using bounday values also rather than relying exclusively on some pop culture curiosity in official testing.